test/jdk/java/awt/TrayIcon/RemoveSystemTrayTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2006, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTException;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.SystemTray;
+import java.awt.TrayIcon;
+import java.awt.image.BufferedImage;
+
+import jtreg.SkippedException;
+import sun.awt.AppContext;
+import sun.awt.SunToolkit;
+
+/*
+ * @test
+ * @bug 6390934
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @modules java.desktop/sun.awt
+ * @summary To check if TrayIcon msg is displayed when there is no system tray.
+ * @run main/manual RemoveSystemTrayTest
+ */
+
+public class RemoveSystemTrayTest {
+    private static SystemTray tray;
+    private static TrayIcon icon;
+    private static final ThreadGroup threadGroup
+            = new ThreadGroup("Test thread group");
+    private static AppContext appContext = null;
+    private static final Object lock = new Object();
+    // stores information whether AppContext was created to prevent spurious wakeups
+    private static volatile boolean isCreated;
+
+    private static final String INSTRUCTIONS = """
+                1) When the test starts an icon is added to the SystemTray area.
+                2) Remove the SystemTray manually (if it's impossible just press PASS).
+                3) If you still see the tray icon in the tray area region
+                   then press FAIL else goto next step.
+                4) Press button 'Show message'.
+                5) If you see the tray icon message then press FAIL else press PASS.
+                """;
```

Comment:
This test was originally created for Solaris issue. 

On macOS and Windows I was unable to remove the "System Tray" although there are options like pause notifications and hide trayicons, but in both cases with tray icon showing as well as hidden I wasn't able to see TrayIco msg.

On Ubuntu I'm unable to remove the System Tray but used the Do Not Disturb option to pause any notifications but in both cases the TrayIcon msg is displayed. 

Is this test relevant to be retained or am I missing something while testing  because I didn't find any options to remove the SystemTray on macOS, Windows & Ubuntu?

Reply:
It's possible on Ubuntu, but not out of the box, it requires disabling the Ubuntu AppIndicators Gnome Shell extension.
AFAIK to disable extensions you need to install additional software, so I suggest just not adding this test.

Reply:
Thank you for clarifying @azvegint.  Deleted the test.
When I checked there were few suggestion online about additional installations required for the removal of tray icon and it didn't seem to be straight-forward (like enabling/disabling Settings) hence wanted to clarify before deleting this test.